src/Tools/ExternalAccess/Razor/RazorCSharpFormattingInteractionService.cs

```diff
@@ -75,19 +75,26 @@ public static SyntaxNode Format(
             CancellationToken cancellationToken)
         {
             Contract.ThrowIfFalse(root.Language is LanguageNames.CSharp);
-            return Formatter.Format(root, services.SolutionServices, GetFormattingOptions(indentationOptions), cancellationToken: cancellationToken);
+            return Formatter.Format(root, services.SolutionServices, GetFormattingOptions(services.SolutionServices, indentationOptions), cancellationToken: cancellationToken);
         }
 
-        private static SyntaxFormattingOptions GetFormattingOptions(RazorIndentationOptions indentationOptions)
-            => new CSharpSyntaxFormattingOptions()
+        private static SyntaxFormattingOptions GetFormattingOptions(SolutionServices services, RazorIndentationOptions indentationOptions)
+        {
+            var legacyOptionsService = services.GetService<ILegacyGlobalOptionsWorkspaceService>();
+            var formattingOptions = legacyOptionsService is null
```

Comment:
Either way this won't work in cohosting. Options are not synced pro-actively to Roslyn OOP, each feature pulls what it needs via callbacks, so even if this service is available the underlying `IGlobalOptionsService` has only defaults in OOP. Callbacks didn't work for Razor services last time I tried (strange IVT-type issues with service hub assembly loading) so this will have to be solved separately.

Sadly cohosting makes some things worse, then we have to go and make them better again.